ddtrace/tracer/tracer_test.go

```diff
@@ -1326,6 +1326,80 @@ func cpspan(s *span) *span {
 	}
 }
 
+type testTraceWriter struct {
+	mu      sync.RWMutex
+	buf     []*span
+	flushed []*span
+}
+
+func newTestTraceWriter() *testTraceWriter {
+	return &testTraceWriter{
+		buf:     []*span{},
+		flushed: []*span{},
+	}
+}
+
+func (w *testTraceWriter) add(spans []*span) {
+	w.mu.Lock()
+	w.buf = append(w.buf, spans...)
+	w.mu.Unlock()
+}
+
+func (w *testTraceWriter) flush() {
+	w.mu.Lock()
```

Comment:
nit: might be a good idea to put a `time.Sleep(10*time.Millsecond)` here to increase the chance of observing the initial race condition with a greater chance.

Reply:
True, but I don't want to customise this struct specifically for one test. It may be used by other tests later and it would seem out of place to me. 🤔